api/v1beta2/types.go

```diff
@@ -261,8 +261,31 @@ type Instance struct {
 	// CapacityReservationID specifies the target Capacity Reservation into which the instance should be launched.
 	// +optional
 	CapacityReservationID *string `json:"capacityReservationId,omitempty"`
+
+	// MarketType specifies the type of market for the EC2 instance. Valid values include:
+	// "on-demand" (default): The instance runs as a standard On-Demand instance.
+	// "spot": The instance runs as a Spot instance. When SpotMarketOptions is provided, the MarketType defaults to "spot".
+	// "capacity-block": The instance utilizes pre-purchased compute capacity (capacity blocks) with AWS Capacity Reservations.
+	//  If this value is selected, CapacityReservationID must be specified to identify the target reservation.
+	// If MarketType is not specified and SpotMarketOptions is provided, the MarketType defaults to "spot".
+	// +optional
+	MarketType *MarketType `json:"marketType,omitempty"`
 }
 
+// MarketType describes the market type of an Instance
+type MarketType string
```

Comment:
You'll need a `kubebuilder:validation:Enum` marker on here to make sure the correct schema is generated